src/application/common/logger.ts

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable, unmanaged } from 'inversify';
+import { LoggerWatcher } from './logger-watcher';
+
+export namespace LogLevel {
+    export const FATAL = 60;
+    export const ERROR = 50;
+    export const WARN = 40;
+    export const INFO = 30;
+    export const DEBUG = 20;
+    export const TRACE = 10;
+}
+
+export const ILoggerServer = Symbol("ILoggerServer");
+
+export interface ILoggerServer {
+    setClient(client: ILoggerClient | undefined): void;
+    setLogLevel(id: number, logLevel: number): Promise<void>;
+    getLogLevel(id: number): Promise<number>;
+    log(id: number, logLevel: number, message: string, params: any[]): Promise<void>;
+    child(obj: Object): Promise<number>;
```

Comment:
What are you trying to achieve by using `Object` type? Prohibit primitives? Then you should use `object`. `Object` is a common type for all types including primitives. I think you want `child(obj: any): Promise<number>` or `child<T extends object>(obj: T): Promise<number>`